pkg/workflows/management/upgrade_curated_packages.go

```diff
@@ -0,0 +1,64 @@
+package management
+
+import (
+	"context"
+
+	appsv1 "k8s.io/api/apps/v1"
+
+	"github.com/aws/eks-anywhere/pkg/constants"
+	"github.com/aws/eks-anywhere/pkg/task"
+	"github.com/aws/eks-anywhere/pkg/utils/ptr"
+	"github.com/aws/eks-anywhere/pkg/workflows"
+)
+
+type upgradeCuratedPackagesTask struct{}
+
+func (s *upgradeCuratedPackagesTask) Run(ctx context.Context, commandContext *task.CommandContext) task.Task {
+	commandContext.PackageInstaller.InstallCuratedPackages(ctx)
+
+	client, err := commandContext.ClientFactory.BuildClientFromKubeconfig(commandContext.ManagementCluster.KubeconfigFile)
+	if err != nil {
+		commandContext.SetError(err)
+	}
+
+	packagesManager := &appsv1.Deployment{}
+
+	err = client.Get(ctx, "eks-anywhere-packages", constants.EksaPackagesName, packagesManager)
+	if err != nil {
+		commandContext.SetError(err)
+		return &workflows.CollectMgmtClusterDiagnosticsTask{}
+	}
+
+	replicas := packagesManager.Spec.Replicas
+	packagesManager.Spec.Replicas = ptr.Int32(0)
+	err = client.Update(ctx, packagesManager)
+	if err != nil {
+		commandContext.SetError(err)
+		return &workflows.CollectMgmtClusterDiagnosticsTask{}
+	}
+
+	packagesManager.Spec.Replicas = replicas
+
+	packagesManager.SetUID("")
+	packagesManager.SetResourceVersion("")
+
+	err = client.Update(ctx, packagesManager)
```

Comment:
Since we only want to update once field, could we just use a `Patch`?

Doing an update without a resource version can lead to race condition where two clients ovewrite each other changes. It's not a big deal since this is not happening in a loop, but I feel a patch would be better.